File: c-tokens/source/escape_token.c
#include "escape_token.h"

// Escape Tokens
const EscapeToken APOSTROPHE = '\'';
const EscapeToken BACKSLASH = '\\';
const EscapeToken CARRIAGE_RETURN = '\r';
const EscapeToken NEW_LINE = '\n';
const EscapeToken NULL_CHARACTER = '\0';
const EscapeToken QUOTATION_MARK = '\"';
const EscapeToken TAB = '\t';

// Escape Token Vector
const EscapeToken escapes_vector() {
    const EscapeToken strings[7] = {
        APOSTROPHE,
        BACKSLASH,
        CARRIAGE_RETURN,
        NEW_LINE,
        NULL_CHARACTER,
        QUOTATION_MARK,
        TAB,
    };

    return *strings;
}

File: tokens/src/escape_token.rs
use std::{
    primitive::char,
    vec::Vec,
};

// Escape Tokens
pub const APOSTROPHE: char = '\'';
pub const BACKSLASH: char = '\\';
pub const CARRIAGE_RETURN: char = '\r';
pub const NEW_LINE: char = '\n';
pub const NULL_CHARACTER: char = '\0';
pub const QUOTATION_MARK: char = '\"';
pub const TAB: char = '\t';

// Escape Token Vector
pub fn escapes_vector() -> Vec<char> {
    let strings: Vec<char> = Vec::from([
        APOSTROPHE,
        BACKSLASH,
        CARRIAGE_RETURN,
        NEW_LINE,
        NULL_CHARACTER,
        QUOTATION_MARK,
        TAB,
    ]);

    return strings;
}
